### PR TITLE
[ADD] general/digest_emails: add new digest emails docs

### PR DESCRIPTION
**Pull request by StraubCreative**
**Overview**
Project Task [here](https://www.odoo.com/web#id=2582128&cids=3&model=project.task&view_type=form)
Requested by @LNAodoo 

**Summary of changes**
This PR adds a new **Digest Emails** section inside **General / Miscellaneous**, and includes 3 articles nested as follows:
- Digest Emails (Overview)
    - Customize _Your Odoo Periodic Digest_
    - Create your own Digest Email*

I also alphabetized the **Miscellaneous** TOC menu in the sidebar to make it easier to navigate so tagging you @jcs-odoo to review.

**(*)Note:** the third article is a work in progress however what's there now was what was on the previous info tab inside Periodic Digest settings (screenshot [here](https://www.awesomescreenshot.com/image/18730805?key=c9c302fe1c0a50a7e7ff72cd3764c8ec) showing old layout). Will continue to work on this with feedback from UX team.

### CODE REVIEW

**Comment by AntoineVDV** [comment]
Hi, @StraubCreative no need to manually assign reviewers. We now have a "codeowners" feature to do that automatically ;)

Codeowners only request a review from @odoo/doc-review when they all submitted an approving review.

File: content/applications/general/digest_emails/customize_periodic_digest.rst
Comment (jcs-odoo) [inline content/applications/general/digest_emails/customize_periodic_digest.rst:5]: Here's a suggestion to illustrate how you could write this more concisely:

```suggestion
To edit a **Digest Email**, go to :menuselection:`Settings --> General Settings --> Statistics -->
Digest Email`, and click on the *external link* next to the dropdown selection list.
```

However, you probably won't use this as what this does is merely open the selected Digest Email currently in use.

If you squash your "customize" and "create" sections, you'll probably explain only once how to open the **Digest Emails** menu from where admins can either open or edit digest emails.

**Review by jcs-odoo — CHANGES_REQUESTED** [review]
Hello @StraubCreative !

That's a nice first one on this repo, I must say! :)
You used plenty of commands specific to this documentation repo.

I added some guideline comments and suggestions to achieve a better result, but it's mainly about the structure and RST rather than content.

Structure-wise:
- Fabien asked us not to put "important" content in the category pages, contrary to what we started to do. That is why you see a few pages in the "misc" section with content in the category pages. It turns out it is not a good example anymore. (sadly. I preferred it this way)
- I think the content of this PR could easily be "squashed" on a single page.
  - This would solve the point above, about the category page, since it wouldn't be a category page anymore
  - It also solves the issue about the titles (see comment in the content itself) 
  - It also helps achieve a better structure inside each page(see comment in the content itself)
  - from a content perspective, it makes sense to split into various pages if the topics are quite different, but here it is not, imo. 
  - Customization and Creation aren't different topics. They both are about the same kind of entries and data fields
    - However, **custom KPI with Odoo Studio** is a different topic

So, I would suggest squashing everything you explain in a single doc. To know what kind of content to explain where and how, please refer to the [content guidelines: structure section](https://www.odoo.com/documentation/15.0/contributing/documentation/content_guidelines.html#document-structure).

Remember that for each heading, there is first a "what it is" explanation (if needed), then a "how to do it."

Here is _a_ suggestion for the structure. I can be different, of course, but maybe this can help you.

- H1: **Digest Emails**
  - "what" explanation of what they are (as you did in digest_emails.rst)
- H2: **Configuration**
  - How to enable/disable the feature (still what's in digest_emails.rst)
  - then how to configure a mail digest (which squashes your "create" and "customize). it could be a different heading, maybe
- H3: **Custom KPIs with Studio**
- H2: ? user side's usage? (probably not necessary, though)

RST wise:
- [break the line at 100th character](https://www.odoo.com/documentation/15.0/contributing/documentation/rst_guidelines.html#use-relative-links-for-internal-urls)
- make sure to leave a white line between each command. Headings and paragraphs are different commands, so a white line is required between them too.
- see comments in the content

Now, you also mention that the KPI part is still a WIP. Do you intend to update this PR before merging it? If so, you can put this PR as a draft until ready for merging-reviewing. But maybe you plan to merge this first and improve the content later? Then it's fine like it is now.

It's already a lot of feedback, but I haven't read the content itself in-depth, so there's not much feedback about it.

Good luck! And let me know if something needs more clarification.

File: content/applications/general/digest_emails/create_digest_email.rst
Comment (jcs-odoo) [reply content/applications/general/digest_emails/create_digest_email.rst:9]: Hi again!

Additional note, now that I think of it! :)

For the link to documentation specific to custom KPIs, you can add an internal hyperlink target such as the suggestion here to have an anchor that won't change, even if you edit the title in RST later.

```suggestion
.. _digest-emails/custom-kpi::

Custom KPIs with Studio
=======================

```
the URL would be ----> ..../general/digest_emails.html#digest-email-custom-kpi. The devs could then use that link in the module itself rather than to the top of the page.

File: content/applications/general.rst
Comment (StraubCreative) [reply content/applications/general.rst:16]: Hi @jcs-odoo 

Thank you for all of the suggestions. I'll work on these and get another version pushed up this week 👍 

As for the reorganization piece, I see where you're coming from. 

I'd like to entertain a separate PR in the future that attempts to reorganize all of the sections alphabetically, with the assumption that this type of organization would be helpful in eliminating friction for the end user (e.g., there's already friction in digesting/implementing the information in the documentation without the added burden of taking the time to search for the correct information).

I see two progressive options we could pursue to produce a _fair_ + _ideal_ working model.
- The _fair_ model uses  "business English" as the leading language as this is the default language set for `/`.
- The _ideal_ model might leverage an extension or custom script inside `conf.py` to accommodate for separate languages, organizing by the first character in the title (something like [this](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#confval-autodoc_member_order) maybe).

Can circle back on it later if you're open to either of these ⏳

Comment (jcs-odoo) [reply content/applications/general.rst:16]: Hello @StraubCreative and @jubodoo 

As I mentioned in my previous comment, I agree that a reorganization is needed, but I also think that a better reorganization would be _outside_ the misc section. I don't have any good suggestion for this right now, though.

I'm not against sorting the categories of the misc section alphabetically, but that would be the only place in the doc where that logic could work, and only for business English, as you pointed out. If you do so, it wouldn't be in this commit about the new page "digest emails" and certainly in another branch based on 13.0 rather than 15.0.

The issue here is that it is a "misc" section. Because there are many different things thrown together in that section, there is no reason to have one category or page before another. So indeed, sorting the misc categories alphabetically could partially work here for that level, but it would also put more "noise" in the structure. For example, the pages 
- In-App Purchase (IAP)
- Developer Mode (debug mode)
were meant to be at the end of the misc section because they are mostly referenced in other doc pages or in the apps themselves. We don't expect users to actually browse the doc structure looking for these two pages.

Also, it wouldn't work per se for the children pages of these categories. For example, here would be the alphabetical order of the children pages of misc/users. "Manage users" which is meant to be the first page of this category would come last:
- Users & Companies
  - Access Rights
  - Change language
  - Companies
  - Manage users

In my opinion, an alphabetical order would certainly not work for the other sections of the documentation.
Take this Websites section for example.
image
Sorting everything in alphabetical order would be a non-sense as "Website" would come last, and the order of the eCommerce categories would make no sense either: 

- Websites
  - eCommerce
    - Collect taxes
    - Get paid
    - Get started
    - Launch my website
    - Manage my products
    - Maximize my revenue
  - Live Chat
  - Website

That's why I commented that, in my opinion, it is not a good answer to the problem, even just for the misc section. I'm not blocking it either, of course. At least you have my detailed opinion :) 

Have a good one!

Jonathan

**Comment by StraubCreative** [comment]
Hi @jcs-odoo !

Here's a 2nd pass @ [6ffe57c](https://github.com/odoo/documentation/pull/1446/commits/6ffe57c224df3c471d268a8230f41594a7d902af) where I've added in all of your suggestions, namely:

- Squashed 3 RST files into one --> `digest_emails.rst`
- Reformatted doc to break at or before 100th character
- Added internal hyperlink anchors where applicable
- Restored sidebar navigation order as it was prior before alphabetizing
- Resized images to 1080p instead of 4k HD
- Content changes to headings and copy per your notes

Looking forward to your review, thanks!

cc: @jubodoo

**Comment by StraubCreative** [comment]
@jcs-odoo hi there! Checking to see if you had a chance to review the most recent push after your change requests.
Thanks and please let me know 🙏 

cc: @jubodoo @AntoineVDV

File: content/applications/general/digest_emails.rst
Comment (jcs-odoo) [inline content/applications/general/digest_emails.rst:53]: Verbs such as "create", "manage" etc are not often necessary.
https://www.odoo.com/documentation/saas-15.1/contributing/documentation/content_guidelines.html#titles-and-headings

Don't forget to add a blank line after the title. Actually, between all different commands. Please adapt the rest of the doc accordingly.

```suggestion
Custom digest emails
====================

```

File: content/applications/general/digest_emails.rst
Comment (jcs-odoo) [inline content/applications/general/digest_emails.rst:71]:
```suggestion
In order to create additional fields, you must create two fields on the digest object:
```

**Review by jcs-odoo — CHANGES_REQUESTED** [review]
Thanks for all the work, Zach.

Here are still a few comments. Some are just ideas.

I skim-read the second half as I have to go, but overall, it's good.


- images
  - ideally take the screenshots with an HD resolution screen, not more. periodic-digest.png is too large.
  - periodic-digest.png looks a bit weird
  - digest-email-settings.png looks like taken on an older version than 15.0 (or the colors are washed out somehow) (nitpicking :) )
  - please compress your pngs with pngquant https://www.odoo.com/documentation/saas-15.1/contributing/documentation.html#contributing-pngquant

**Comment by StraubCreative** [comment]
@jcs-odoo 

Thank you for the review today. 
Here's the latest batch of changes as reflected on [a7ac065](https://github.com/odoo/documentation/pull/1446/commits/a7ac0655408dca5915a95441aaff9625ac3b7db1):

- fixed remaining 100th character breaks
- new screenshots for `periodic-digest.png` and `digest-email-settings.png`, taken at 110% zoom, sized to 1080p + compressed with pngquant
- note indentation fixed
- typo at line 36 fixed
- rephrased lines 36-38 per suggestion
- removed future tense from doc, changed to present
- Removed “create” from heading at line 54, added blank line after all headings in doc
- line 74, number to word change
- removed ordinal language and semicolons in ordered list starting on line 78

Please let me know if there's anything else or if we're good to merge, thanks again! 🙏 

cc: @jubodoo

File: content/applications/general/digest_emails.rst
Comment (jcs-odoo) [reply content/applications/general/digest_emails.rst:12]: As explained in the main comment, the line breaks suggested here were
- line 7 after ``high-level``
- line 10 after ``Statistics`,``

File: content/applications/general/digest_emails.rst
Comment (jcs-odoo) [inline content/applications/general/digest_emails.rst:63]:
```suggestion
After you click **Save**, your new custom Digest Email is available as a selection in the **General
Settings** dropdown menu.
```

File: content/applications/general/digest_emails.rst
Comment (jcs-odoo) [inline content/applications/general/digest_emails.rst:38]:
```suggestion
To customize the default Digest Email (*Your Odoo Periodic Digest*), go to :menuselection:`Settings 
--> General Settings --> Statistics --> Digest Email`, select *Your Odoo Periodic Digest* and click
on the *external link* next to the dropdown selection.
```

**Review by jcs-odoo — APPROVED** [review]
Hi @StraubCreative :) 
Thanks: here are some last comments, but I have already approved the PR.

Since I approved it, you can ask for a review from @odoo/doc-review . They're the team that checks the PR's technical part and then merges it. So, with my reviews, it should help you get your PR approved by them ;)

- the line break still isn't right.
  - you can go to the line everywhere that there is a space. So that can be "in the middle" of a `menuselection` or a bold text.
  - In my suggestion at lines 7-12, GitHub displayed it as follow, but even if "high-level" appears on another line, it's still part of line 7 
    image

- about the images, I didn't mean that they all have to be 1080 pixels wide. Apologies :) I meant on an HD (1080p) screen. That gives a good size for the documentation. But it works well like that now, so no worries.
- Usually, please click on "resolve conversation" yourself if you agree with the last comment and that you applied it.


Cheers :)

**Comment by StraubCreative** [comment]
@jcs-odoo thanks again for all of your feedback on this PR 🙏 

I went ahead and applied your last suggestions for 100th character cleanup on lines 7, 10, 36, and 62 as seen on [bab2a36](https://github.com/odoo/documentation/pull/1446/commits/bab2a36a68e5f288504f40bb32717beb74cd105e). 

In all honesty it might take me a few more tries to get right but I appreciate the guidance around these little nuances-- for example on line 7 my text editor says that's 101 characters but it seems like you're saying _not_ to count markup symbols like *, `, etc. in the character count?

In any case, think we're ready for technical review. 
Thanks again!

File: content/applications/general/digest_emails.rst
Comment (AntoineVDV) [inline content/applications/general/digest_emails.rst:1]: This metadata is used to make the page accessible when it contains a `toctree` directive.
```suggestion
```

File: content/applications/general/digest_emails.rst
Comment (AntoineVDV) [inline content/applications/general/digest_emails.rst:56]: I'm not sure if mentioning the arrow here is worth it. It might be removed/changed with another symbol in Odoo and require updating this page.
```suggestion
To do so, click on **Configure Digest Emails** and then **Create**.
```

File: content/applications/general/digest_emails.rst
Comment (AntoineVDV) [inline content/applications/general/digest_emails.rst:86]: Separate the headers from the rows.
```suggestion
+=======================+=========================================+
```

File: content/applications/general/digest_emails.rst
Comment (AntoineVDV) [inline content/applications/general/digest_emails.rst:87]: This might be cleaner to highlight the values as "technical terms".
```suggestion
| Connected Users       | `kpi_res_users_connected_value`         |
```

**Review by AntoineVDV — CHANGES_REQUESTED** [review]
Hi @StraubCreative, please find my review comments below :)

> my text editor says that's 101 characters but it seems like you're saying not to count markup symbols like *, `, etc. in the character count?

Every single character counts. Make sure your editor is a proper text editor (not a Word-like soft) and that it's properly configured (it should be the case by default).

**Comment by StraubCreative** [comment]
@AntoineVDV thanks for your feedback 🙏 
Went ahead and committed your suggestions which can be seen on [052034f](https://github.com/odoo/documentation/pull/1446/commits/052034f08ac69a7e670accaf9a79467d683669d1).

- Removed unnecessary `show-toc` metadata on line 1
- Removed "->" arrow on/near line 56
- Added backticks (`) to technical values in ref table.
- ALSO: fixed incorrect Value for **New Leads** Label in ref table (was a duplicate Value of row 2 before).

Please let me know if there are any other items you'd like me to address.
Thanks again for all of your help. Excited!

**Review by AntoineVDV — APPROVED** [review]
LGTM!

@robodoo r+

**Comment by jcs-odoo** [comment]
👌

**Pull request closed without merging**
